### ISSUE TITLE
API for changing the configuration settings

### ISSUE THREAD

**Issue by ajaymandke** [ccrisan/motioneye#96]
First of all, great program!. I am using Ubuntu build on Raspberry pi and its working great. I have Motioneye v0.24.

Is there API available for changing the configuration?

**Comment by ccrisan** [comment]
No, there is no such api (yet). Also, you should upgrade to 0.30.

**Comment by ajaymandke** [comment]
Thanks. I was able to upgrade to 0.30 after some initial issues. Thanks. The new interface and options (about google drive upload etc.) look great.

**Comment by JamBalaya56562** [comment]
Short answer for current motionEye: there is still no separately documented, stable public configuration API, but the web UI itself does use JSON endpoints internally.

For example, the current UI reads and writes configuration through endpoints like:

- `GET /config/main/get/`
- `GET /config/<camera_id>/get/`
- `POST /config/main/set/`
- `POST /config/<camera_id>/set/`
- `POST /config/0/set/` for multiple camera updates

These require admin authentication and the JSON payload is the same shape that the web UI sends, so I would treat them as internal implementation details rather than a guaranteed external API. If you automate against them, the safest approach is to first read the current config with `.../get/`, modify only the fields you need, then send the updated JSON back with `.../set/`.

For simple runtime actions, such as pausing/resuming detection or triggering snapshots, the motion HTTP control API or motionEye action endpoints are usually a better fit than editing the full camera config.

This issue is from the old 0.24/0.30 era and the internals have changed a lot since then, so if someone still needs a documented public configuration API today, a fresh issue with the exact automation use case would be the best way to scope it.

**Comment by MichaIng** [comment]
Right. Maybe someone finds time to document the API in our [wiki](https://github.com/motioneye-project/motioneye/wiki), including the new login loop.

Literally and naturally everything the GUI does can be done manually via API, even with more flexibility, and some actions can be done via `meyectl` CLI.

I'll mark this old question as closed.
